internal/jobcontainers/path.go

```diff
@@ -0,0 +1,143 @@
+package jobcontainers
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+func searchPathForExe(fileName, pathsToSearch string) (string, error) {
+	fileNamePtr, err := windows.UTF16PtrFromString(fileName)
+	if err != nil {
+		return "", err
+	}
+
+	pathsToSearchPtr, err := windows.UTF16PtrFromString(pathsToSearch)
+	if err != nil {
+		return "", err
+	}
+
+	extension, err := windows.UTF16PtrFromString(".exe")
+	if err != nil {
+		return "", err
+	}
+
+	path := make([]uint16, windows.MAX_PATH)
+	_, err = winapi.SearchPath(
+		pathsToSearchPtr,
+		fileNamePtr,
+		extension,
+		windows.MAX_PATH,
+		&path[0],
+		nil,
+	)
+	if err != nil {
+		return "", err
+	}
+
+	exePath := windows.UTF16PtrToString(&path[0])
+	// Need to check if we just found a directory with the name of the executable and
+	// .exe at the end. ping.exe is a perfectly valid directory name for example.
+	attrs, err := os.Stat(exePath)
+	if err != nil {
+		return "", &os.PathError{Path: exePath, Op: "stat", Err: err}
```

Comment:
`Stat` already returns a `PathError` is there some reason we wrap that?

Reply:
Good catch, none